composer/callbacks/checkpoint_saver.py

```diff
@@ -358,48 +391,54 @@ def _save_checkpoint(self, state: State, logger: Logger, log_level: LogLevel):
 
         is_deepspeed = is_model_deepspeed(state.model)
 
-        if is_deepspeed and '{rank}' not in self.filename.filename:
-            raise ValueError(f'Save filename {self.filename.filename} must have {{rank}} for deepspeed.')
+        if is_deepspeed and '{rank}' not in self.file_path.filename:
+            raise ValueError(f'Save filename {self.file_path.filename} must have {{rank}} for deepspeed.')
 
         # save the checkpoint to the filename
-        filename = self.filename.format(state, is_deepspeed)
+        file_path = self.file_path.format(state, is_deepspeed)
 
         saved_path = checkpoint.save_checkpoint(
             state=state,
-            filename=filename,
+            filename=file_path,
             weights_only=self.weights_only,
         )
 
         if not saved_path:  # not all ranks save
             return
 
+        # Create symlink to latest checkpoint file.
         if self.latest_filename is not None:
-            symlink = self.latest_filename.format(state, is_deepspeed)
+            symlink = self.latest_file_path.format(state, is_deepspeed)
             os.makedirs(os.path.dirname(symlink), exist_ok=True)
             try:
                 os.remove(symlink)
             except FileNotFoundError:
                 pass
-            os.symlink(os.path.relpath(filename, os.path.dirname(symlink)), symlink)
+            os.symlink(os.path.relpath(file_path, os.path.dirname(symlink)), symlink)
 
-        # if artifact name provided, upload the checkpoint
-        if self.artifact_name is not None:
-            artifact_name = self.artifact_name.format(
-                state,
-                is_deepspeed,
-            ).lstrip('/')
+        # If upload_file is True, then user specified a folder argument like so:
+        # scheme://{bucket}/{path}, which means the artifact name and the file_path are the same.
+        if self.upload_file:
+            # Upload checkpoint file
+            artifact_name = file_path
+            if self.using_wandb:
+                artifact_name = artifact_name.replace('/', '.')
```

Comment:
same question as other places, but why does this replace happen? Maybe add a comment if it is something specific to what wandb expects